README.md

```diff
@@ -188,149 +476,514 @@ Options:
-npm run build
-node build/cli.js [command] [options]
-```
+## Measure Logic Highlighting
 
-Optionally, you can install the `ts-node` utility globally to execute the TypeScript files directly instead of running the build script:
+`fqm-execution` can provide highlighted HTML to indicate which individual pieces of the measure logic CQL had "truthy" or "falsy" values. Generation of this HTML is enabled by default, and can be disabled by setting the `calculateHTML` calculation option to `false`.
 
-```bash
-npm install -g ts-node
-ts-node src/cli.ts [command] [options]
+The following example shows a scenario where the logic highlighting for a patient indicates that they are not in the numerator population of the measure due to the timing constraints of a FHIR `Procedure` resource that exists on their record.
+
+![Screenshot of Highlighted HTML Measure Logic](./static/logic-highlighting-example.png)
+
+In detailed results calculation, an HTML string will appear on each `detailedResults` object for a given patient:
+
+```typescript
+import { Calculator } from 'fqm-execution';
+
+// ...
+
+const { results } = await Calculator.calculate(measureBundle, patientBundles, {
+  calculateHTML: true /* true by default, but explicitly shown for demonstration */
+});
+
+// results
+/*    ^?
+      [
+          {
+            "patientId": "patient-1",
+            "detailedResults": [
+              {
+                "groupId": "group-1",
+                "html": ""<div><h2>Population Group: group-1</h2> ..."
+                ...
+              }
+              ...
+            ]
+            ...
+          }
+          ...
+     ]
+*/
+```
+
+In `MeasureReport` calculation, an HTML string will appear on the `.text.div` property of the returned FHIR `MeasureReport` for a given patient:
+
+```typescript
+import { Calculator } from 'fqm-execution';
+
+// ...
+
+const { results } = await Calculator.calculateMeasureReports(measureBundle, patientBundles, {
+  calculateHTML: true /* true by default, but explicitly shown for demonstration */,
+  reportType: 'individual' /* highlighting is only available for individual MeasureReport calculation */
+});
+
+// results
+/*    ^?
+        [
+          {
+            "id": "<some-random-uuid>",
+            "resourceType": "MeasureReport",
+            "text": {
+              "status": "generated",
+              "div": "<div><h2>Population Group: group-1</h2> ..."
+            }
+            ...
+          }
+          ...
+        ]
+ */
 ```
 
-### Debug Option
+**NOTE**: Measure logic highlighting is not available when calculating a population `MeasureReport` with `reportType = 'summary'`
+
+## Group Clause Coverage Highlighting
+
+`fqm-execution` can generated highlighted HTML that indicates which individual pieces of the measure logic CQL were processed at all during calculation, regardless of whether or not they held "truthy" or "falsy" value. This is often referred to as "Clause Coverage". "Covered" clauses will
```

Comment:
```suggestion
`fqm-execution` can generate highlighted HTML that indicates which individual pieces of the measure logic CQL were processed at all during calculation, regardless of whether or not they held "truthy" or "falsy" values. This is often referred to as "Clause Coverage". "Covered" clauses will
```

Reply:
You could tell I was losing steam at this point

Reply:
For the amount of text in this PR, the lack of grammar/spelling errors is genuinely baffling.